jkube-kit/enricher/specific/src/test/java/org/eclipse/jkube/kit/enricher/specific/PrometheusEnricherTest.java

```diff
@@ -75,59 +71,62 @@ public void testCustomPrometheusPort() {
-                .build())
-            .build());
-    final KubernetesListBuilder builder = new KubernetesListBuilder().withItems(
-        new ServiceBuilder().withNewMetadata().withName("foo").endMetadata().build()
-    );
-    // When
-    prometheusEnricher.create(PlatformMode.kubernetes, builder);
-    // Then
-    assertThat(builder.buildFirstItem().getMetadata().getAnnotations())
-        .hasSize(3)
-        .containsEntry("prometheus.io/port", "9779")
-        .containsEntry("prometheus.io/scrape", "true")
-        .containsEntry("prometheus.io/path", "/metrics");
+      // Given
+      initContext(null,
+              ImageConfiguration.builder().build(
+                              BuildConfiguration.builder()
+                                      .ports(Arrays.asList("1337", null, " ", "9779", null))
+                                      .build())
+                      .build());
+      final KubernetesListBuilder builder = new KubernetesListBuilder().withItems(
+              new ServiceBuilder().withNewMetadata().withName("foo").endMetadata().build()
+      );
+      PrometheusEnricher prometheusEnricher = new PrometheusEnricher(context);
+      // When
+      prometheusEnricher.create(PlatformMode.kubernetes, builder);
+      // Then
+      assertThat(builder.buildFirstItem().getMetadata().getAnnotations())
+              .hasSize(3)
+              .containsEntry("prometheus.io/port", "9779")
+              .containsEntry("prometheus.io/scrape", "true")
+              .containsEntry("prometheus.io/path", "/metrics");
   }
 
   @Test
   public void testNoDefinedPrometheusPort() {
-    // Given
-    initContext(null,
-        ImageConfiguration.builder().build(
-            BuildConfiguration.builder()
-                .ports(Collections.emptyList())
-                .build())
-            .build());
-    final KubernetesListBuilder builder = new KubernetesListBuilder().withItems(
-        new ServiceBuilder().withNewMetadata().withName("foo").endMetadata().build()
-    );
-    // When
-    prometheusEnricher.create(PlatformMode.kubernetes, builder);
-    // Then
-    assertThat(builder.buildFirstItem().getMetadata().getAnnotations()).isNull();
+      // Given
+      initContext(null,
+              ImageConfiguration.builder().build(
+                              BuildConfiguration.builder()
+                                      .ports(Collections.emptyList())
+                                      .build())
+                      .build());
+      final KubernetesListBuilder builder = new KubernetesListBuilder().withItems(
+              new ServiceBuilder().withNewMetadata().withName("foo").endMetadata().build()
+      );
+      PrometheusEnricher prometheusEnricher = new PrometheusEnricher(context);
+      // When
+      prometheusEnricher.create(PlatformMode.kubernetes, builder);
+      // Then
+      assertThat(builder.buildFirstItem().getMetadata().getAnnotations()).isNull();
```

Comment:
I'm wondering how this assertion still works with Kubernetes Client 6.0.

Please, replace `isNull()` with `isNullOrEmpty()` (there was a change in the Kubernetes Client 6.0 `ObjectMeta` class, this assertion is more appropriate).